### ISSUE TITLE
Freezes on :x on large file

### ISSUE THREAD

**Issue by rgoulter** [helix-editor/helix#4309]
### Summary

When using `helix` to open large files (> 1.0MiB), when I try to run `:x` or `:wq` the editor hangs.

Using `:w` `:q` instead behaves as I'd expect.

### Reproduction Steps

 

I tried this:

- For a big file, I was using `~/.local/share/fish/fish_history`. But using e.g. plaintext from gutenberg https://www.gutenberg.org/ebooks/1342

```
wget https://www.gutenberg.org/files/1342/1342-0.txt -O pnp.txt
cat pnp.txt pnp.txt > pnp2.txt
du -sh pnp2.txt
```

1. open with `hx pnp2.txt`

2. invoke command `:x`.

I expected this to happen:

The editor to quit. I expect that `wq` should behave the same as `:w` `:q`.

Instead, this happened:

The editor is unresponsive, and looks as it does before `<Enter>` is hit.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

with `-vvv`,

e.g. when it hangs:
```
2022-10-16T19:21:23.083 helix_loader [DEBUG] Located configuration folders: []
2022-10-16T19:21:23.088 helix_view::clipboard [INFO] Using xclip to interact with the system and selection (primary) clipboard
2022-10-16T19:21:23.098 mio::poll [TRACE] registering event source with poller: token=Token(1), interests=READABLE | WRITABLE
2022-10-16T19:21:23.098 mio::poll [TRACE] registering event source with poller: token=Token(2), interests=READABLE | WRITABLE
2022-10-16T19:21:23.098 mio::poll [TRACE] registering event source with poller: token=Token(0), interests=READABLE
2022-10-16T19:21:23.098 mio::poll [TRACE] registering event source with poller: token=Token(1), interests=READABLE
2022-10-16T19:21:24.781 helix_term::job [DEBUG] waiting on jobs...
```

e.g. when using `:w` `:q`:
```
2022-10-16T19:22:06.326 helix_loader [DEBUG] Located configuration folders: []
2022-10-16T19:22:06.331 helix_view::clipboard [INFO] Using xclip to interact with the system and selection (primary) clipboard
2022-10-16T19:22:06.333 mio::poll [TRACE] registering event source with poller: token=Token(1), interests=READABLE | WRITABLE
2022-10-16T19:22:06.333 mio::poll [TRACE] registering event source with poller: token=Token(2), interests=READABLE | WRITABLE
2022-10-16T19:22:06.333 mio::poll [TRACE] registering event source with poller: token=Token(0), interests=READABLE
2022-10-16T19:22:06.333 mio::poll [TRACE] registering event source with poller: token=Token(1), interests=READABLE
2022-10-16T19:22:07.205 helix_term::job [DEBUG] waiting on jobs...
2022-10-16T19:22:07.207 helix_term::job [DEBUG] waiting on jobs...
2022-10-16T19:22:07.208 mio::poll [TRACE] deregistering event source from poller
2022-10-16T19:22:07.208 mio::poll [TRACE] deregistering event source from poller
2022-10-16T19:22:07.208 mio::poll [TRACE] deregistering event source from poller
```

</details>

### Platform

Linux & MacOS

### Terminal Emulator

kitty 0.26.2, & macOS Terminal

### Helix Version

helix 22.08.1 (66276ce6)

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/3975
